src/core/actix/inbox.rs

```diff
@@ -0,0 +1,50 @@
+use crate::{
+    core::object_id::ObjectId,
+    data::Data,
+    traits::{ActivityHandler, Actor, ApubObject},
+    Error,
+    LocalInstance,
+};
+
+use crate::core::actix::signature::verify_signature;
+use actix_web::{dev::Payload, FromRequest, HttpRequest, HttpResponse};
+use http_signature_normalization_actix::prelude::DigestVerified;
+use serde::de::DeserializeOwned;
+use tracing::debug;
+
+/// Receive an activity and perform some basic checks, including HTTP signature verification.
+pub async fn receive_activity<Activity, ActorT, Datatype>(
+    request: HttpRequest,
+    activity: Activity,
+    local_instance: &LocalInstance,
+    data: &Data<Datatype>,
+) -> Result<HttpResponse, <Activity as ActivityHandler>::Error>
+where
+    Activity: ActivityHandler<DataType = Datatype> + DeserializeOwned + Send + 'static,
+    ActorT: ApubObject<DataType = Datatype> + Actor + Send + 'static,
+    for<'de2> <ActorT as ApubObject>::ApubType: serde::Deserialize<'de2>,
+    <Activity as ActivityHandler>::Error: From<anyhow::Error>
+        + From<Error>
+        + From<<ActorT as ApubObject>::Error>
+        + From<serde_json::Error>
+        + From<http_signature_normalization_actix::digest::middleware::VerifyError>,
+    <ActorT as ApubObject>::Error: From<Error> + From<anyhow::Error>,
+{
+    // ensure that payload hash was checked against digest header by middleware
+    DigestVerified::from_request(&request, &mut Payload::None).await?;
```

Comment:
This line isnt can be removed now as we are verifying the digest directly in the lib now. Maybe `http-signature-normalization-actix` dependency can also be removed then.

Reply:
Removing it would be great due to its licensing. That dependency is very restrictive about the license. (freedom of licenses of they are compatible would be nice but they limit it to 2 choices)